processor/Cargo.toml

```diff
@@ -23,6 +23,7 @@ std = ["vm-core/std", "winterfell/std", "log/std"]
 log = "0.4.14"
 vm-core = { package = "miden-core", path = "../core", version = "0.4", default-features = false }
 winterfell = { package = "winter-prover", version = "0.4.2", default-features = false }
+hex = { version = "0.4" , default-features = false , features = ["alloc"]}
```

Comment:
I'm wondering if instead of doing this, we should provide a function in `miden-core` [utils](https://github.com/0xPolygonMiden/miden-vm/blob/next/core/src/utils/mod.rs). We already use do custom conversion to hex (e.g., [here](https://github.com/0xPolygonMiden/miden-vm/blob/next/core/src/program/blocks/call_block.rs#L86-L89)) - and might be good to handle this conversion in a centralized place.

The function could be something like:
```Rust
pub fn write_hex(f: &mut core::fmt::Formatter<'_>, bytes: &[u8]) -> core::fmt::Result
```

Reply:
The reason why I used `hex` is because that is already used in the codebase in the CLI, for [example](https://github.com/0xPolygonMiden/miden-vm/blob/9c8af67a7768b920772b3a42d89947f7f639e089/miden/src/cli/run.rs#L38)

Is the goal to remove those too?

Reply:
> The reason why I used `hex` is because that is already used in the codebase in the CLI, for [example](https://github.com/0xPolygonMiden/miden-vm/blob/9c8af67a7768b920772b3a42d89947f7f639e089/miden/src/cli/run.rs#L38)

In the `miden-vm` crate we are using a few extra dependencies, but in other crates we've been trying to reduce the number of dependencies to the minimum required. Would be good to remove extra dependencies from `miden-vm` too - but we could do it at a later point.

Reply:
Instead of calling `write!` in a loop, which can be very inefficient depending on the underlying writer having buffer or not, I allocated a String with the right size and wrote to it. That made it simpler to have `to_hex -> Result<String, Error>` instead.

Let me know if that is good enough.

Reply:
Yep - I like this better. Not so much because of efficiency reasons, but because I think the function which returns a string is easier to use.